File: src/main/java/bazooka/client/data/ShooterData.java
package bazooka.client.data;

import javax.persistence.*;
import java.io.Serializable;

@Entity(name = "Shooter")
public class ShooterData implements Serializable {

  @Id @GeneratedValue
  Integer id;

  @Column(unique = true)
  String name;

  public ShooterData() {}

  public ShooterData(String name) {
    this.name = name;
  }

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  @Override public boolean equals(Object obj) {
    if (obj instanceof ShooterData) {
      ShooterData other = (ShooterData) obj;
      return id != null && id.equals(other.id);
    }
    return false;
  }

  @Override public int hashCode() {
    return 6502 + id ^ (id >>> 32);
  }

  @Override public String toString() {
    return name;
  }
}

File: src/main/scala/bazooka/server/data/ShooterData.scala
package bazooka.server.data

import javax.persistence._
import org.apache.commons.lang.builder._

@Entity{val name = "Shooter"}
class ShooterData(shooterName: String) {
  
  @Id
  @GeneratedValue
  var id: java.lang.Integer = null

  @Column{val unique = true}
  var name: String = shooterName

  @Lob
  var script: String = null

  def this() = this("")

  override def equals(obj: Any): Boolean = {
    val other = obj.asInstanceOf[ShooterData]
    new EqualsBuilder()
      .append(this.id, other.id)
      .isEquals()
  }

  override def hashCode(): Int = {
    new HashCodeBuilder()
      .append(id)
      .toHashCode
  }

  override def toString(): String = name
}